core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -479,6 +484,23 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging {
     getAll.sorted.map{case (k, v) => k + "=" + v}.mkString("\n")
   }
 
+  def getDriverExtraClassPath() : Option[String] = {
+    // Concatenate common and driver classpath in that order
+    val commonExtraClassPath = getOption("spark.common.extraClassPath")
+    val driverExtraClassPath = getOption("spark.driver.extraClassPath")
+    Option((commonExtraClassPath ++ driverExtraClassPath).mkString(File.pathSeparator)).filter(_
```

Comment:
Actually, it can be null, right? If neither spark.common.extraClassPath nor spark.driver.extraClassPath are set, I'd like this function to return None so I am not changing the original semantics of anybody calling this function. And, if so, casting as option is right thing, no?

If there is a more legible way of expressing this, I am totally open to it.